pkg/services/ngalert/eval/eval.go

```diff
@@ -207,19 +202,6 @@ func getExprRequest(ctx AlertExecCtx, data []models.AlertQuery, now time.Time, d
 			datasources[q.DatasourceUID] = ds
 		}
 
-		// If the datasource has been configured with custom HTTP headers
-		// then we need to add these to the request
-		decryptedData, err := secretsService.DecryptJsonData(ctx.Ctx, ds.SecureJsonData)
```

Comment:
This code was added (#44862) because middleware *wasn't* setting required custom headers on prometheus datasource queries.

How can we verify that this change doesn't break that behavior? The custom headers middleware is older than the change to explicitly set custom headers here, and it's not clear that the datasource plugins will always pass requests through the middleware.

My *guess* is that this change fixes the broken headers behavior at a more fundamental level than the code you're removing, but I can't tell if we're explicitly verifying that: https://github.com/grafana/grafana/commit/3df34fe06452d7a5c5369b27d81514497a57485b

That change also only applies to the Prometheus data source - each datasource, including external plugins, needs to be sure to use the httpclientprovider's client in order to get the custom headers from the plugin settings. Is that right?

Reply:
> How can we verify that this change doesn't break that behavior? The custom headers middleware is older than the change to explicitly set custom headers here, and it's not clear that the datasource plugins will always pass requests through the middleware.

I just pushed some integration tests for querying Prometheus via Grafana HTTP API. Something I've had in mind for long time, but haven't figured out how to do it until now.

> My guess is that this change fixes the broken headers behavior at a more fundamental level than the code you're removing, but I can't tell if we're explicitly verifying that

Yes, via HTTPClientProvider and verified with mentioned integration test.

> That change also only applies to the Prometheus data source - each datasource, including external plugins, needs to be sure to use the httpclientprovider's client in order to get the custom headers from the plugin settings. Is that right?

Yes, that's correct

Reply:
@aocenas FYI :point_up: